### PR TITLE
Add "environment:" host config to update /etc/environment

### LINKED ISSUE #587: Pharos cluster does not support no_proxy configuration
In our environment all outgoing addresses require proxy. There is our own private docker repository, which is not accessible with proxy.

Working solution is to modify on all cluster nodes: 
manually add /etc/system.d/system/docker.d/http-proxy.conf to have NO_PROXY-option 

> Environment="NO_PROXY=repository.server.local" 

Also modifying  /etc/environment to contain following line 

> NO_PROXY="repository.server.local" 

It would be nice to have cluster.yml a option to setup the no-proxy-settings, due we need to verify the settings every time we run 
> pharos up -c cluster.yml 

Affects the pharos 1.3.0 -version

### PR DESCRIPTION
**Pull request by kke**
Fixes #587 

Removes the host `proxy:` configuration and replaces it with a generic "environment:" configuration for updating `/etc/environment`.

When setting the `http_proxy`, you should usually also set the uppercase `HTTP_PROXY` since some programs use one and some the other.

This also enables the use of `NO_PROXY` as requested in #587 and separately resolved in #611 

To remove an entry from `/etc/environment` set a null value:

```yaml
hosts:
  - address: 127.0.0.1
     environment:
       PATH: null # or leave blank
```

### CODE REVIEW

File: lib/pharos/host/configurer.rb [hunk 1]
```diff
@@ -96,6 +96,35 @@ def cluster_config
         self.class.cluster_config
       end
 
+      # @return [Pharos::SSH::File]
+      def env_file
+        @ssh.file('/etc/environment')
+      end
+
+      def update_env_file
+        return if @host.environment.nil? || @host.environment.empty?
```
Comment (jakolehm) [inline lib/pharos/host/configurer.rb]: Is this one-way thing? One can add envs but never remove them?

Comment (kke) [reply lib/pharos/host/configurer.rb]: Yes. If the `original_data.merge(@host.environment).compact.map` below would be changed to something that would not overwrite `nil`'s during merge, you could delete by setting `environment: FOOFOO: null`.

Comment (kke) [reply lib/pharos/host/configurer.rb]: Now changed to do so.

```ruby
{ a: nil, b: 'foo' }.merge(a: 'foo', c: 'bar') { |a,b| b }
=> {:a=>nil, :b=>"foo", :c=>"bar"}
{ a: nil, b: 'foo' }.merge(a: 'foo', c: 'bar') { |a,b| b }.compact
=> {:b=>"foo", :c=>"bar"}
```

Comment (jakolehm) [reply lib/pharos/host/configurer.rb]: Specs would be nice.

Comment (kke) [reply lib/pharos/host/configurer.rb]: Whoomp, there it is.

**Pull request merged**